src/keboola_mcp_server/tools/oauth.py

```diff
@@ -0,0 +1,64 @@
+"""OAuth URL generation tools for the MCP server."""
+
+import logging
+from typing import Annotated
+
+from fastmcp import Context
+from fastmcp.tools import FunctionTool
+from pydantic import Field
+
+from keboola_mcp_server.client import KeboolaClient
+from keboola_mcp_server.errors import tool_errors
+from keboola_mcp_server.mcp import KeboolaMcpServer, with_session_state
+
+LOG = logging.getLogger(__name__)
+
+TOOL_GROUP_NAME = 'OAUTH'
+
+
+def add_oauth_tools(mcp: KeboolaMcpServer) -> None:
+    """Adds OAuth tools to the MCP server."""
+    mcp.add_tool(FunctionTool.from_function(create_oauth_url))
+    LOG.info('OAuth tools added to the MCP server.')
+
+
+@tool_errors()
+@with_session_state()
+async def create_oauth_url(
+    component_id: Annotated[
+        str, Field(description='The component ID to grant access to (e.g., "keboola.ex-google-analytics-v4").')
+    ],
+    config_id: Annotated[str, Field(description='The configuration ID for the component.')],
+    ctx: Context,
+) -> str:
+    """
+    Generates an OAuth authorization URL for a Keboola component configuration.
+
+    When using this tool, be very concise in your response. Just guide the user to click the
+    authorization link.
+
+    Note that this tool should be called specifically for these OAuth-requiring components after their
```

Comment:
```suggestion
    Note that this tool should be called specifically for the OAuth-requiring components after their
    configuration is created e.g. keboola.ex-google-analytics-v4 and keboola.ex-gmail.
```
OR if you want to emphasize those two components
```suggestion
    Note that this tool should be called specifically for these OAuth-requiring components after their
    configuration is created: keboola.ex-google-analytics-v4 and keboola.ex-gmail.
```